AnkiDroid/src/main/java/com/ichi2/utils/AdaptionUtil.java

```diff
@@ -0,0 +1,23 @@
+package com.ichi2.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.os.Build;
+
+import java.util.List;
+
+public class AdaptionUtil {
+    public static boolean hasWebBrowser(Context context){
```

Comment:
```suggestion
    private static boolean sHasRunWebBrowserCheck = false;
    private static boolean sHasWebBrowser = true;

    public static boolean hasWebBrowser(Context context) {
        if (sHasRunWebBrowserCheck) {
            return sHasWebBrowser;
        }
        
```

Since this is static and should not vary much, probably best to memoize this as it could affect performance. I haven't tested this suggested code but it should work or be close to working